### ISSUE TITLE
Open file from cmdline in existing instance got slower with v1.53

### ISSUE THREAD

**Issue by mbaas2** [microsoft/vscode#116771]
Issue Type: <b>Bug</b>

VSCode is typically open all day here, but I change workspaces etc. from within  the running instance - and sometimes I do open files from the commandline using `code -a {file}`. This seems to have slowed down significantly after updating to the January-release.

VS Code version: Code 1.53.2 (622cb03f7e070a9670c94bae1a45d78d7181fbd4, 2021-02-11T11:48:04.245Z)
OS version: Windows_NT x64 10.0.19041

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i9-8950HK CPU @ 2.90GHz (12 x 2904)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: enabled<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.72GB (3.66GB free)|
|Process Argv|-a C:\\devt\\TACIT\\Jenkins\\GitHub.apln --crash-reporter-id 9a9701f0-2aca-4268-b8fa-342fffdd67fe|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (60)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|13.0.1
pascal|ale|9.2.0
pascal-formatter|ale|2.4.0
vscode-json|and|1.5.0
vscode-bad-code-highlighter|bha|0.1.6
jsonviewer|cci|1.2.2
bracket-pair-colorizer-2|Coe|0.2.0
vscode-markdownlint|Dav|0.38.0
githistory|don|0.6.14
xml|Dot|2.5.1
dotup-vscode-diagnostics-search|dot|1.0.0
gitlens|eam|11.2.1
vscode-powertools|ego|0.64.0
auto-close-tag|for|0.5.10
auto-rename-tag|for|0.1.6
code-runner|for|0.11.2
vscode-pull-request-github|Git|0.23.1
todo-tree|Gru|0.0.196
beautify|Hoo|1.5.0
output-colorizer|IBM|0.1.2
bridlensis|idl|1.2.1
innosetup|idl|1.6.0
nsis|idl|3.38.1
nsis-plugins|idl|1.3.0
nsl-assembler|idl|1.2.0
search-editor-apply-changes|jak|0.1.1
csharpextensions|jch|1.3.1
svg|joc|1.4.5
statusbarerror|Joe|0.4.0
vscode-position|jtr|1.0.1
csharpfixformat|Leo|0.0.84
markdown-shortcuts|mdi|0.12.0
rainbow-csv|mec|1.8.1
code-point|med|1.7.4
xml-format|mik|1.1.2
vscode-edits-history|mis|0.1.6
vscode-json5|mrm|1.0.0
vscode-docker|ms-|1.9.1
csharp|ms-|1.23.9
vscode-edge-devtools|ms-|1.1.3
python|ms-|2021.1.502429796
jupyter|ms-|2020.12.414227025
powershell|ms-|2020.6.0
debugger-for-edge|msj|1.0.15
favorites|ole|1.5.1
vscode-apl-language|Opt|0.0.7
vscode-apl-language-client|Opt|0.0.10
vscode-xyplorer|Oti|0.2.1
open-html-in-browser|pea|2.1.3
tsimporter|pmn|2.0.1
MDTools|sea|1.0.1
code-settings-sync|Sha|3.4.3
vscode-fileutils|sle|3.4.4
pdf|tom|1.1.0
highlight-matching-tag|vin|0.10.0
vscode-icons|vsc|11.1.0
vscode-todo-highlight|way|1.0.4
local-history|xyz|1.8.1
markdown-authoring|yan|0.1.5
vscode-preview-server|yui|1.3.0

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstry244:30256636
pythonvsdeb440:30248342
pythonvsded773:30248341
pythonvspyt875:30251590

```

</details>

**Comment by bpasero** [comment]
Can you follow this wiki and post the results back here to understand more about why performance is so bad for you: https://github.com/Microsoft/vscode/wiki/Performance-Issues

Thanks!

**Comment by mbaas2** [comment]
Will do - once GitHub have sorted their issue displaying that wiki:
![image](https://user-images.githubusercontent.com/5645494/108079156-c7755100-706e-11eb-8716-119b8af4351a.png)
(tried with current Chome,  Edge & Firefox)

**Comment by jrieken** [comment]
Can you try this:  `code --prof-startup -a {file} ` and then follow the prompt and attach the profiles here. Thanks

**Comment by mbaas2** [comment]
Done. I tried 2 scenarios: 

1. code not running:
[prof-9907.extHost.cpuprofile.txt](https://github.com/microsoft/vscode/files/5989484/prof-9907.extHost.cpuprofile.txt)
[prof-9907.renderer.cpuprofile.txt](https://github.com/microsoft/vscode/files/5989485/prof-9907.renderer.cpuprofile.txt)
[prof-9907.main.cpuprofile.txt](https://github.com/microsoft/vscode/files/5989490/prof-9907.main.cpuprofile.txt)

2. editing one file and repeating the excercise - no prompts and only this file (w/o .txt-extension):
[prof-cc99.txt](https://github.com/microsoft/vscode/files/5989515/prof-cc99.txt)

**Comment by jrieken** [comment]
> which is a pity, because scenario 2 is my most common one.

Yeah, the startup profiler needs a blank state. From looking at the attached profiles from scenario 1: I don't see any slow there, did you experience a slow start when you ran this?

**Comment by mbaas2** [comment]
That's all subjective, now ofc - but it felt significantly slower than 1.52
Especially "refreshing the edit pane" seems to be affected, I also see this when task-switching back to code...until the cursor appears...(but that should probably be it's own issue) ...

**Comment by mbaas2** [comment]
Hmm, is there anywhere I can download older versions just to compare the timing?

**Comment by bpasero** [comment]
Yes, here is `1.53` as standalone zip: https://az764295.vo.msecnd.net/stable/ea3859d4ba2f3e577a159bc91e3074c5d85c0523/VSCode-win32-x64-1.52.1.zip

**Comment by mbaas2** [comment]
Ah, thanks! So, aftzer comparing these 2, I must revise my report, it seems. The real issue is when attempting to open additional files using the "-a"-switch. While loading code & opening the first file takes approx. 4seconds, opening a 2nd file can takes 30 secs or more. Code gets focus really fast, but then it takes a loon time until the additional tab appears and is filled with content. and gets a blinking cursor. v1.52 is significantly faster at that!
(I tested both on the same files which all were < 5k)

**Comment by bpasero** [comment]
@mbaas2 `-a` is to add a folder to a window, not for opening files.

**Comment by mbaas2** [comment]
Opps...  (I was using a cmd in my filemgr to open code, and I had setup that link to "open with code" while ago - and I forgot about the meaning of these switches.)  I have now updated and only use -a when a folder is passed.

I played around a bit, looking for a repro, I timed and compared a scenario that is quite close to my uses cases - and it clearly showed poor performance. But unfortunately that did not repeat when I tried a 2nd time. I'll keep looking if I can come up with a better repro and may open another issue when I have one.

Thanks!
